tensorflow_datasets/image_classification/image_folder.py

```diff
@@ -29,147 +29,72 @@
 
 
 SUPPORTED_IMAGE_FORMAT = (".jpg", ".jpeg", ".png")
+DATA_DIR = os.path.join('~', 'tensorflow_datasets')
+AUTOTUNE = tf.data.experimental.AUTOTUNE
 
 
-class ImageLabelFolder(tfds.core.GeneratorBasedBuilder):
-  """Generic image classification dataset created from manual directory.
-
-  The data directory should have the following structure:
-
-  ```
-  path/to/manual_dir/<dataset_name>/
-    split_name/  # Ex: 'train'
-      label1/  # Ex: 'airplane' or '0015'
-        xxx.png
-        xxy.png
-        xxz.png
-      label2/
-        xxx.png
-        xxy.png
-        xxz.png
-    split_name/  # Ex: 'test'
-      ...
-  ```
-
-  To use it:
-
-  ```
-  builder = tfds.image.ImageLabelFolder('<dataset_name>')
-  dl_config = tfds.download.DownloadConfig(manual_dir='path/to/manual_dir/')
-  builder.download_and_prepare(download_config=dl_config)
-  print(builder.info)  # Splits, num examples,... automatically extracted
-  ds = builder.as_dataset(split='split_name', shuffle_files=True)
-  ```
-
-  Or with load:
-
-  ```
-  dl_config = tfds.download.DownloadConfig(manual_dir='path/to/manual_dir/')
-  tfds.load(
-      'image_label_folder',
-      split='split_name'
-      builder_kwargs=dict(dataset_name='<dataset_name>'),
-      download_and_prepare_kwargs=dict(download_config=dl_config),
-  )
-  ```
-
-  """
-
-  MANUAL_DOWNLOAD_INSTRUCTIONS = "This is a 'template' dataset."
-
-  VERSION = tfds.core.Version(
-      "2.0.0", "New split API (https://tensorflow.org/datasets/splits)")
-
-  # TODO(epot): Image shape should be automatically deduced
-
-  def __init__(self, dataset_name="image_label_folder", **kwargs):
-    self.name = dataset_name
-    super(ImageLabelFolder, self).__init__(**kwargs)
-
-  def _info(self):
-    if not self._data_dir:
-      logging.warning(
-          "ImageLabelFolder.info is only complete once the data has been "
-          "generated. Please call .download_and_prepare() before calling "
-          ".info. The .info.features won't be computed.")
-
-    return tfds.core.DatasetInfo(
-        builder=self,
-        description="Generic image classification dataset.",
-        # Generic features before the data is generated
-        features=tfds.features.FeaturesDict({
-            "image": tfds.features.Image(),
-            "label": tfds.features.ClassLabel(num_classes=None),
-        }),
-        supervised_keys=("image", "label"),
-    )
-
-  def _split_generators(self, dl_manager):
-    """Returns SplitGenerators from the folder names."""
-    # At data creation time, parse the folder to deduce number of splits,
-    # labels, image size,
-
-    # The splits correspond to the high level folders
-    split_names = list_folders(dl_manager.manual_dir)
-
-    # Extract all label names and associated images
-    split_label_images = {}  # dict[split_name][label_name] = list(img_paths)
+class ImageLabelFolder():
```

Comment:
After thinking more about it, it might be best to have `ImageFolder` inherit from `tfds.core.GeneratorBasedBuilder` but we should list all caveat/pain point to make this works properly and found the best implementation:

* It shouldn't be added to the register (maybe useing the `registered.skip_registration` ?).
* The __init__ signature should be different (only contains `data_dir=`)
* Data dir shouldn't be `DATA_DIR/dataset_name/config/version` but the given version.
* `download_and_prepare` should raise an error if called.
* `DatasetInfo`  fields & cie are dynamically computed in `__init__`
* Other ?

For the implementation, it seems there are three options:

1. Do not inherit at all from `DatasetBuilder`
2. Inherit from `DatasetBuilder` and hack around the API.
3. Split DatasetBuilder in two class `DatasetBase` and `DatasetBuilder`

Without trying, it's difficult to know which one of those solutions is the best. My feeling would be to try 2 first and see what are the hack required (overwrite `__init__`,...). If it is really too hard, eventually prototype what would need to split `DatasetBase` and `DatasetBuilder` in 2 class and have `DatasetBuilder(DatasetBase)` and `ImageFolder(DatasetBase)`.

Reply:
Just for clarification, should I try this first 
* `ImageFolder` inherit from `tfds.core.GeneratorBasedBuilder`
or this
* Inherit from `DatasetBuilder` and hack around the API?

Reply:
I meant inherit from `DatasetBuilder`, not `GeneratorBasedBuilder`. Sorry for the confusion. I updated the text

Reply:
> Data dir shouldn't be DATA_DIR/dataset_name/config/version but the given version.

Data dir should be `DATA_DIR/version` or `DATA_DIR` or something else?

Reply:
Data dir should be the path where the images are located. There is no version.
If images are located in: `path/to/images/train/img0001.jpeg`, then data_dir should be `path/to/images/`

Reply:
Yes, there shouldn't be any copy. Just read the plain images.